### ISSUE TITLE
Eddy fails when bvec contains an entry with all zeros

### ISSUE THREAD

**Issue by bjeurissen** [MRtrix3/mrtrix3#421]
On some of my data sets, non diffusion weighted images are encoded as 0 0 0 0 in the gradient table. 

This seems to upset recent versions of eddy (segmentation fault), which apparently are expecting all bvecs to have norm 1.

This is probably more a bug of FSL, but for now it might be good to enforce norm one vectors in the distcorr scripts.

**Comment by Lestropie** [comment]
Alternatively, should this be caught at DICOM import (lib/file/dicom/image.cpp:452)?

**Comment by jdtournier** [comment]
Well, that won't catch existing images, or images converted using other packages... I think this should be handled in the script...

**Comment by Lestropie** [comment]
The other option is whether null-norm vectors should be caught and changed to [0 0 1] in `save_bvecs_bvals()`. That way the change for keeping FSL happy extends beyond those scripts.

**Comment by jdtournier** [comment]
Sure, I'd be happy enough with that. Personally I think we shouldn't need to do this, and I don't like the fact that the encoding will be different after conversion to FSL and back, but I can't think of any circumstances where the change would be anything but benign... Feel free to use that approach if it makes life easier.

**Comment by bjeurissen** [comment]
Pretty sure this wasn't an issue with fsl 5.0.7. But both 5.0.8 and 5.0.9 choke on it. Probably normalises with a zero norm or something.

Think the save_bvecs_bvals() approach makes sense and is a pragmatic solution. I agree that it should not be necessary, but that the effects should be benign.

**Comment by Lestropie** [comment]
I think eddy had an issue with [0 0 0] vectors with the least-squares reconstruction in earlier versions: Would look for matching reversed-PE volumes based on a dot product. But it was never a problem because there were other issues that prevented LSR from being used; most notably not being supported (and reporting as such after fixing the aforementioned bug).

Maybe a relevant question is whether [0 0 0] vectors may cause issues anywhere else in FSL? If they do, then the `save_bvecs_bvals()` modification makes more sense; if it's purely `eddy`, then the script approach preserves that reversibility (even if the absence of such should be benign). Maybe not the right forum to ask though.

**Comment by jdtournier** [comment]
Yes, I remember this popped up a while ago. I think on balance we might as well output the FSL vectors as [0 0 1] - it's so unlikely to cause any problem for anything else...
